lib/collection.dart

```diff
@@ -77,3 +77,23 @@ bool setsEqual(Set a, Set b) {
 
   return a.containsAll(b);
 }
+
+/// Returns the index of the first item in [elements] where [predicate]
+/// evaluates to true.
+///
+/// Returns -1 if there are no items where [predicate] evaluates to true.
+int indexOf<T>(Iterable<T> elements, bool predicate(T element)) {
+  if (elements is List) {
+    for (int i = 0; i < elements.length; i++) {
+      if (predicate(elements[i])) return i;
```

Comment:
Iterable doesn't have `operator[]`, so you'll need a cast. I'd extract a local.
```dart
if (elements is List) {
  var elementList = elements as List;
  for (int i = 0; i < elementList.length; i++) {
    if (predicate(elements[i])) return i;
  }
}

...
```

Alternatively, stick with the original code. @yjbanov thoughts?